lib/awful/placement.lua

```diff
@@ -1422,6 +1422,11 @@ function placement.restore(d, args)
 
     if not memento then return false end
 
+    -- Apparently, 0x0 did happen to at least 1 person. Officially can't
+    -- happen, but can't be too careful.
+    if memento.width  and memento.width < 1 then return end
+    if memento.height and memento.height < 1 then return end
```

Comment:
How about moving this into the creation of a memento instead, so that such an invalid memento cannot be created at all?

However, I am curious how this could happen. This seems to require `c:geometry()` to return 0 for width/height. This can only happen while the client is still being managed, because there is a time frame while the value is still 0 before the values coming from the X11 server are applied. This can only happen during the following signals: `new`, `property::x`, `property::y`, `property::width`, `property::screen`, `property::window`.

An (in my opinion) even better proposal: Instead of working around this in Lua, change the C code so that it emits the signals later (except for `new`, but no one should use that one anyway)?
Basically:
```diff
diff --git a/objects/client.c b/objects/client.c
index ec4ab424b..9f6bd138b 100644
--- a/objects/client.c
+++ b/objects/client.c
@@ -1405,8 +1405,6 @@ client_manage(xcb_window_t w, xcb_get_geometry_reply_t *wgeom, xcb_get_window_at
      * (Else, reparent could cause an UnmapNotify) */
     xcb_change_window_attributes(globalconf.connection, w, XCB_CW_EVENT_MASK, select_input_val);
 
-    luaA_object_emit_signal(L, -1, "property::window", 0);
-
     /* The frame window gets the border, not the real client window */
     xcb_configure_window(globalconf.connection, w,
                          XCB_CONFIG_WINDOW_BORDER_WIDTH,
@@ -1429,15 +1427,16 @@ client_manage(xcb_window_t w, xcb_get_geometry_reply_t *wgeom, xcb_get_window_at
 
     /* Store initial geometry and emits signals so we inform that geometry have
      * been set. */
-#define HANDLE_GEOM(attr) \
-    c->geometry.attr = wgeom->attr; \
-    luaA_object_emit_signal(L, -1, "property::" #attr, 0);
-HANDLE_GEOM(x)
-HANDLE_GEOM(y)
-HANDLE_GEOM(width)
-HANDLE_GEOM(height)
-#undef HANDLE_GEOM
-
+    c->geometry.x = wgeom->x;
+    c->geometry.y = wgeom->y;
+    c->geometry.width = wgeom->width;
+    c->geometry.height = wgeom->height;
+
+    luaA_object_emit_signal(L, -1, "property::x", 0);
+    luaA_object_emit_signal(L, -1, "property::y", 0);
+    luaA_object_emit_signal(L, -1, "property::width", 0);
+    luaA_object_emit_signal(L, -1, "property::height", 0);
+    luaA_object_emit_signal(L, -1, "property::window", 0);
     luaA_object_emit_signal(L, -1, "property::geometry", 0);
 
     /* Set border width */
```